[WIP] pf5

MariaAga opened the pull request:
Updating pf4 to pf5 
some work is done by running

npx @patternfly/pf-codemods@latest webpack --fix

search& replace "pf-c-" to "pf-v5-c-"

Check all "onChange=\{[^(]" to make sure that pure pf5 are using the right arguments, as they changed order, and most onChange are onChange(event,value) and not onChange(value)

And adding act wrapper around tests.
To test this pr you will need the foreman core pf5 pr, foreman js pr, link foreman-js, and to remove any foreman-js/patternfly folder in the katello node_modules (as this breaks React shared context in tests).

This is a WIP since the other prs were just done and not polished, and also because some of the tests I had troubles fixing and dont have capacity currently to fix. These tests fail with ` Error: Uncaught [Error: Warning: An update to %s inside a test was not wrapped in act(...)` even if every line in the test is wrapped in act, or
```
 Error: Error: Nock: No match for request {
      "method": "GET",
      "url": "http://localhost/api/v2/hosts/1/subscriptions/available_release_versions",
      ...
```

MariaAga commented:
Fixed more tests but not sure what to do about the last 8 failing with this type of error as in some of them I do see the request being mocked in the test
```
    Error: Error: Nock: No match for request {
      "method": "GET",
      "url": "http://localhost/api/hosts/test-host",
      "headers": {
        "accept": "application/json, text/plain, */*",
        "x-requested-with": "XMLHttpRequest",
        "x-csrf-token": "",
        "referer": "http://localhost/",
        "user-agent": "Mozilla/5.0 (linux) AppleWebKit/537.36 (KHTML, like Gecko) jsdom/11.12.0",
        "accept-language": "en",
        "host": "localhost",
        "accept-encoding": "gzip, deflate"
      }
    }
```

sjha4 commented on webpack/scenes/AlternateContentSources/Create/__tests__/acsCreate.test.js:
```diff
@@ -387,6 +389,7 @@ test('Can display create wizard and create simplified ACS', async (done) => {
   assertNockRequest(smartProxyScope);
-  assertNockRequest(createScope, done);
+  assertNockRequest(createScope);
+  done();
```
Think we have the act(done) in the tests to stop further API calls from being recorded and for nocks to get cleaned up..I am not entirely sure we need the done() here..I can try locally..

MariaAga replied:
Running `assertNockRequest(X, done);` did not work for tests, I had to change it to:
```
assertNockRequest(X);
done();
```
  In all of them

sjha4 commented:
This should fix the lint errors:

`npx eslint --fix webpack/`

This leaves only some `no-unused-vars` issues in 4 files..

sjha4 commented:
Overall I am not seeing any issues so far in the code/testing.
Minor notes I'll document in comments.

1. Notifications seem a little off: 
PF4:
![Screenshot from 2025-01-03 15-47-21](https://github.com/user-attachments/assets/376ecad1-bd7a-4b16-9dee-d815ed808ee2)
PF5:
![Screenshot from 2025-01-03 15-47-51](https://github.com/user-attachments/assets/0d918b0a-77d4-4b3d-b9a3-19ab4c045822)

2. Some sub-tabs are not rendering full-width:
PF4:
![Screenshot from 2025-01-03 15-51-32](https://github.com/user-attachments/assets/28395e07-ffa9-431d-8d80-378b3667f7a9)
PF5:
![Screenshot from 2025-01-03 15-51-16](https://github.com/user-attachments/assets/12702d1e-e510-4228-83d0-fca562a5b938)

sjha4 commented on webpack/scenes/ContentViews/components/EnvironmentLabels.js:
```diff
@@ -11,15 +11,15 @@ const EnvironmentLabels = (environments) => {
       <React.Fragment key={env.id} style={{ marginBottom: '5px' }}>
         <Label
           color={labelColor}
-          isTruncated
```
Did `isTruncated` get deprecated too?

MariaAga replied:
"Removed the isTruncated property from Label. This is now the default behavior. In addition, you can limit the text width using the new textMaxWidth property."
https://v5-archive.patternfly.org/get-started/upgrade/release-notes

sjha4 commented:
For the conflicts here:

webpack/scenes/ContentViews/Details/Versions/Delete/RemoveSteps/CVReassignActivationKeysForm.js, the is branch code is right.

For webpack/scenes/ContentViews/Details/Versions/Delete/RemoveSteps/CVVersionRemoveReview.js , The master branch has some logic which needs to be maintained.
This looks like the right code with only `isTruncated `removed from label in the master branch code.

```
{singleCVActivationKeysCount > 0 && (
            <Flex>
              <FlexItem><ExclamationTriangleIcon /></FlexItem>
              <FlexItem><p>{__(`${pluralize(singleCVActivationKeysCount, 'activation key')} will be moved to content view ${selectedCVNameForAK} in `)}</p></FlexItem>
              <FlexItem><Label color="purple" href={`/lifecycle_environments/${selectedEnvForAK[0].id}`}>{selectedEnvForAK[0].name}</Label></FlexItem>
            </Flex>
          )}
          {multiCVActivationKeysCount > 0 && (
            <Flex>
              <FlexItem><ExclamationTriangleIcon /></FlexItem>
              <FlexItem>
                <p>
                  {__(`Content view environment will be removed from ${pluralize(multiCVActivationKeysCount, 'multi-environment activation key')}.`)}
                </p>
              </FlexItem>
            </Flex>
          )}
```

MariaAga commented:
Fixed the notifications in foreman, and sub tabs in Katello. Also fixed the lint issues, will resolve the conflicts (once laptop will get unstuck :upside_down_face: )

sjha4 commented:
The new changes work good..Notification is proper and sub-tab tables occupy full width.

Noticed some weirdness in the global registration page, randomly the dropdowns show up like this:
![Screenshot from 2025-01-06 15-54-00](https://github.com/user-attachments/assets/5e9de78f-bef5-47c8-8e6a-32aa28d5bde7)

sjha4 commented:
Locally, I am seeing only 7 failures, some nock ones and others where react test is failing to read by label text on 2 pages..Looking at those..Not sure what's causing the CI failures here though..

MariaAga commented:
CI failures are due to foreman-js pr not being merged/released, so the packages are wrong

MariaAga commented:
Fixed select2 issue in foreman core

sjha4 commented:
The dropdown looks good now..Functionally seems to be in a good state..Looking into tests to see if they can be fixed locally..

sjha4 reviewed (approved):
With the foreman-js, foreman PRs, this PR works for Katello components in as much as I tested. The code looks right..The CI errors here are due to dependency and there are some relevant failures when running locally..I'm ok to merge this and get a look at the actual CI errors when the dependencies are resolved and have a follow-up PR for those.

sjha4 commented:
I have the commit with local test fixes but we need to run the CI once to make sure the same tests fail on the CI with PF5..
https://github.com/sjha4/katello/commit/08aae79ee52203fd3bdd2dad70b6a1e4ae09d7f8

jeremylenz commented:
@MariaAga looks like this needs a rebase, wonder why ;)

sjha4 commented:
Not sure if you included this https://github.com/sjha4/katello/commit/08aae79ee52203fd3bdd2dad70b6a1e4ae09d7f8 into the PR for the test errors?

MariaAga commented:
Yes I included your commit, the new test failures came from the new tests that were written meanwhile. Updated everything so this should be good to go in
